### ISSUE TITLE
Add custom validation rule support

### ISSUE THREAD

**Issue by PWKad** [aurelia/validatejs#4]
https://validatejs.org/#custom-validator

Add decorator / fluent API to cover.

**Comment by apawsey** [comment]
Has any progress been made on this?  Is there a structure that is being proposed?

Quick suggestion would be something in the plugin config that dynamically adds properties to the current validator object, to provide the same api approach with custom validators as built-in ones.  Alternatively, a validators.custom property, that we then extend, if it's preferred to have some kind of division?

I assume the custom validators would need to be configured in the DI somehow, as for example the api lookup type validator mentioned in #29 would need httpClient injected in.

**Comment by PWKad** [comment]
No progress on this one yet PR welcome.  Under the covers the ValidationRule class should probably just have a static method for adding the custom rule that works just like the current rule methods work.

**Comment by aivins** [comment]
@PWKad - I ended up hacking this together for myself today but I'm not sure I took the right approach. validate.async returns a promise, whereas aurelia-validatejs seems to strongly assume the opposite. I ended up converting all validate return values into promises, then using Promise.all() to wait on them all at once before returning to the original caller. I also publish each error into the reporter when Promise.all() resolves.

I also had to patch validate.js's async so that it doesn't reject if an async function returns an error.  That was stopping me from handling it later on with Promise.all().
